### ISSUE TITLE
[Bug]: Error: performing initial clone: repository already exists

### ISSUE THREAD

**Issue by tstraley** [flipt-io/flipt#3710]
### Bug Description

When using git declarative storage backend, and setting `storage.git.backend.type=local` to keep the git repo on the filesystem rather than in memory, it appears that restarting the process can fail because flipt already cloned the repo to the local filesystem location.

```
{"L":"INFO","T":"2024-12-10T16:53:01Z","M":"flipt starting","version":"v1.51.1","commit":"7d46879a4ba113916a3901cec7dbf0189db8fc90","date":"2024-11-05T22:18:52Z","go_version":"go1.23.2"}
Error: performing initial clone: repository already exists
```

### Version Info

v1.51.1  (commit 7d46879a4ba113916a3901cec7dbf0189db8fc90)

Installed via flipt helm chart v0.73.1 in Kubernetes

### Search

- [X] I searched for other open and closed issues before opening this

### Steps to Reproduce

Configure flipt with `storage.git.backend.type=local` and some path eg. `storage.git.backend.path=/tmp/`

Start flipt once, and make sure it successfully clones the git repository and runs.

Restart flipt without removing the git backend data store.

---

In Kubernetes specifically, we have the git backend path writing to an ephemeral volume on the node, and if the container restarts for any reason (eg. liveness probe failure, OOM, etc.) then we end up in crash looping state due to this issue. Simple reproduction can be performed in k8s by setting up the git backend path to an `emptyDir` volume, and then restarting the container with `kubectl exec -it -n flipt <flipt-pod> -- kill 1`.

### Expected Behavior

Flipt should be able to start up successfully, even if the git backend was already cloned into the storage path defined.

i.e. I should be able to restart the flipt service without needing to remove the local git store.

### Additional Context

Full relevant configuration section:

```
storage:
  git:
    backend:
      path: /usr/opt/flipt/git-store
      type: local
    directory: feature-flags
    poll_interval: 30s
    ref: main
    repository: <redacted>
  read_only: true
  type: git
```

**Comment by markphelps** [comment]
Hey @tstraley . Taking a look at this now. Thank you for reporting

**Comment by markphelps** [comment]
@tstraley will create a new release tomorrow with this fix included. Once the nightly build runs (in the next hour) you should be able to try the nightly image and verify the fix https://github.com/flipt-io/flipt?tab=readme-ov-file#nightly-build

**Comment by markphelps** [comment]
@tstraley we just released 1.53.2 that should resolve this issue. please let me know!

**Comment by tstraley** [comment]
Thanks @markphelps - We've rolled out the fix in our environments, and confirmed the issue is no longer reproducible. Thank you so much for the fast resolution.

Happy holidays!
